sdk/export/metric/aggregation/temporality.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate stringer -type=Temporality
+
+package aggregation // import "go.opentelemetry.io/otel/sdk/export/metric/aggregation"
+
+import (
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/sdkapi"
+)
+
+// Temporality indicates the kind of data exported by an exporter.
+// These bits may be OR-d together when multiple exporters are in use.
+type Temporality int
+
+const (
+	// CumulativeTemporality indicates that an Exporter expects a
+	// Cumulative Aggregation.
+	CumulativeTemporality Temporality = 1
+
+	// DeltaTemporality indicates that an Exporter expects a
+	// Delta Aggregation.
+	DeltaTemporality Temporality = 2
+)
+
+// Includes tests whether `kind` includes a specific kind of
+// exporter.
+func (kind Temporality) Includes(has Temporality) bool {
+	return kind&has != 0
+}
+
+// MemoryRequired returns whether an exporter of this kind requires
+// memory to export correctly.
+func (kind Temporality) MemoryRequired(mkind sdkapi.InstrumentKind) bool {
+	switch mkind {
+	case sdkapi.HistogramInstrumentKind, sdkapi.GaugeObserverInstrumentKind,
+		sdkapi.CounterInstrumentKind, sdkapi.UpDownCounterInstrumentKind:
+		// Delta-oriented instruments:
+		return kind.Includes(CumulativeTemporality)
+
+	case sdkapi.CounterObserverInstrumentKind, sdkapi.UpDownCounterObserverInstrumentKind:
+		// Cumulative-oriented instruments:
+		return kind.Includes(DeltaTemporality)
+	}
+	// Something unexpected is happening--we could panic.  This
+	// will become an error when the exporter tries to access a
+	// checkpoint, presumably, so let it be.
+	return false
+}
+
+type (
+	constantTemporalitySelector  Temporality
+	statelessTemporalitySelector struct{}
+)
+
+var (
+	_ TemporalitySelector = constantTemporalitySelector(0)
+	_ TemporalitySelector = statelessTemporalitySelector{}
+)
+
+// ConstantTemporalitySelector returns an TemporalitySelector that returns
+// a constant Temporality, one that is either always cumulative or always delta.
```

Comment:
It will return the passed `Temporality` without changing, there is no grantee the passed `Temporatlity` is only cumulative or delta though.

```suggestion
// a constant Temporality.
```